pkg/api/handlers_node.go

```diff
@@ -376,50 +388,50 @@ func (api *API) ChatPageHandler() gin.HandlerFunc {
 // an HTML page displaying the node's status and uptime info.
 func (api *API) NodeStatusPageHandler() gin.HandlerFunc {
 	return func(c *gin.Context) {
-		peers := api.Node.Host.Network().Peers()
-		nodeData := api.Node.NodeTracker.GetNodeData(api.Node.Host.ID().String())
-		if nodeData == nil {
-			c.HTML(http.StatusOK, "index.html", gin.H{
-				"TotalPeers":       0,
-				"Name":             "Masa Status Page",
-				"PeerID":           api.Node.Host.ID().String(),
-				"IsStaked":         false,
-				"IsTwitterScraper": false,
-				"IsDiscordScraper": false,
-				"IsWebScraper":     false,
-				"FirstJoined":      api.Node.FromUnixTime(time.Now().Unix()),
-				"LastJoined":       api.Node.FromUnixTime(time.Now().Unix()),
-				"CurrentUptime":    "0",
-				"Rewards":          "Coming Soon!",
-				"BytesScraped":     0,
-			})
-			return
-		} else {
-			nd := *nodeData
-			nd.CurrentUptime = nodeData.GetCurrentUptime()
-			nd.AccumulatedUptime = nodeData.GetAccumulatedUptime()
-			nd.CurrentUptimeStr = pubsub.PrettyDuration(nd.CurrentUptime)
-			nd.AccumulatedUptimeStr = pubsub.PrettyDuration(nd.AccumulatedUptime)
-
-			sharedData := db.SharedData{}
-			nv := db.ReadData(api.Node, api.Node.Host.ID().String())
-			_ = json.Unmarshal(nv, &sharedData)
-			bytesScraped, _ := strconv.Atoi(fmt.Sprintf("%v", sharedData["bytesScraped"]))
-			c.HTML(http.StatusOK, "index.html", gin.H{
-				"TotalPeers":       len(peers),
-				"Name":             "Masa Status Page",
-				"PeerID":           api.Node.Host.ID().String(),
-				"IsStaked":         nd.IsStaked,
-				"IsTwitterScraper": nd.IsTwitterScraper,
-				"IsDiscordScraper": nd.IsDiscordScraper,
-				"IsWebScraper":     nd.IsWebScraper,
-				"FirstJoined":      api.Node.FromUnixTime(nd.FirstJoinedUnix),
-				"LastJoined":       api.Node.FromUnixTime(nd.LastJoinedUnix),
-				"CurrentUptime":    nd.CurrentUptimeStr,
-				"TotalUptime":      nd.AccumulatedUptimeStr,
-				"BytesScraped":     fmt.Sprintf("%.4f MB", float64(bytesScraped)/(1024*1024)),
-			})
+		// Initialize default values for the template data
+		templateData := gin.H{
+			"TotalPeers":       0,
+			"Name":             "Masa Status Page",
+			"PeerID":           api.Node.Host.ID().String(),
+			"IsStaked":         false,
+			"IsTwitterScraper": false,
+			"IsDiscordScraper": false,
+			"IsWebScraper":     false,
+			"FirstJoined":      api.Node.FromUnixTime(time.Now().Unix()),
+			"LastJoined":       api.Node.FromUnixTime(time.Now().Unix()),
+			"CurrentUptime":    "0",
+			"TotalUptime":      "0",
+			"Rewards":          "Coming Soon!",
+			"BytesScraped":     "0 MB",
 		}
+
+		if api.Node != nil && api.Node.Host != nil {
```

Comment:
Just a style nitpick which isn't blocking: better to return early when conditions aren't met than adding this under an if closure branch

Otherwise we render `index.html` without templateData